### ISSUE TITLE
[Bug]: Live reload does not work with Netlify starter

### ISSUE THREAD

**Issue by jurajkrivda** [remix-run/remix#557]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [X] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [X] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.0.4

### Steps to Reproduce

Create the app with `npx create-remix@latest` and select Netlify.

### Expected Behavior

Content should be changed

### Actual Behavior

The content has not changed

**Comment by uwuru** [comment]
What commands are you running? Are you running both commands as per the instructions?

It sounds like you are not also running `yarn dev` which deals with the hot reloading.

**Comment by jurajkrivda** [comment]
Hi. Thank you for your response.
My command: `"dev": "concurrently \"remix watch\" \"yarn dev:netlify\"",`
but I tried separately in two CLI windows. 

In Chrome dev console I see 
```
(index):19 WebSocket connection to 'ws://localhost:8002/socket' failed: 
(anonymní) @ (index):19
(index):31 Remix dev asset server web socket error:
```

**Comment by gabrielvincent** [comment]
I'm having the same problem. It happens in both cases:

### Case 1
```
npm run remix watch # on terminal window 1
npm run dev:netlify # on terminal window 2
```

### Case 2

```
npm run remix dev
# if I run `npm run dev:netlify` on another terminal window the command fails because port 3000 is already in use
```

I both cases I see the following error on the browser console:

```
(index):19 WebSocket connection to 'ws://localhost:8002/socket' failed: 
(index):31 Remix dev asset server web socket error:
(index):32 
Event {isTrusted: true, type: 'error', target: WebSocket, currentTarget: WebSocket, eventPhase: 2, …}
isTrusted: true
bubbles: false
cancelBubble: false
cancelable: false
composed: false
currentTarget: WebSocket {url: 'ws://localhost:8002/socket', readyState: 3, bufferedAmount: 0, onopen: null, onerror: ƒ, …}
defaultPrevented: false
eventPhase: 0
path: []
returnValue: true
srcElement: WebSocket {url: 'ws://localhost:8002/socket', readyState: 3, bufferedAmount: 0, onopen: null, onerror: ƒ, …}
target: WebSocket {url: 'ws://localhost:8002/socket', readyState: 3, bufferedAmount: 0, onopen: null, onerror: ƒ, …}
timeStamp: 118.5
type: "error"
[[Prototype]]: Event
```

**Comment by kalijonn** [comment]
while creating the WebSocket Server, remix reads devServerPort from remix.config.js in the root directory. By default it's 8888 while the websocket is trying to connect to 8002. By changing devServerPort on remix.config.js to 8002, LiveReload works. 

My first open source contribution! yay!

P.S: I looked at CONTRIBUTING.md but looks like the Contributor License Agreement is not ready yet.

Regards,
Kali.

**Comment by gabrielvincent** [comment]
Congratultions, @kalijonn! Thank you for finding this fix. 

> P.S: I looked at CONTRIBUTING.md but looks like the Contributor License Agreement is not ready yet.

does this mean that you can’t submit a PR with this fix?

**Comment by kalijonn** [comment]
> Congratultions, @kalijonn! Thank you for finding this fix.
> 
> > P.S: I looked at CONTRIBUTING.md but looks like the Contributor License Agreement is not ready yet.
> 
> does this mean that you can’t submit a PR with this fix?

I'm not sure. I'm new to this and so taking the guidelines very literally. Don't want to create unnecessary headache for anyone. I've forked the repo and I'm in the process of raising a PR. There's a bunch of tests failing (not related to the change I've made). Will try to fix them and get the PR ready. After this, I'll see if the team can guide me.

**Comment by bushblade** [comment]
Came here to post the same bug, can confirm that @kalijonn's solution works for me.
```js
module.exports = {
  appDirectory: 'app',
  browserBuildDirectory: 'public/build',
  publicPath: '/build/',
  serverBuildDirectory: 'netlify/functions/server/build',
  devServerPort: 8002, // should not be 8888
}
```

**Comment by na2hiro** [comment]
I believe this can be closed as the template is fixed and it doesn't _initially_ happen.

I came across this because I changed the port in remix config (I'm running multiple remix apps 🙂). My solution is to pass `port` prop to the component `<LiveReload port={8888} />` in `root.tsx`.

It would be nice if the template is set up so the app reads `remix.config.js` and pass it to LiveReload automatically.
